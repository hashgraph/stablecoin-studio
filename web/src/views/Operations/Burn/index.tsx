import { Heading, Text, Stack, useDisclosure } from '@chakra-ui/react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import DetailsReview from '../../../components/DetailsReview';
import InputNumberController from '../../../components/Form/InputNumberController';
import OperationLayout from '../OperationLayout';
import ModalsHandler from '../../../components/ModalsHandler';
import type { ModalsHandlerActionsProps } from '../../../components/ModalsHandler';
import { useSelector } from 'react-redux';
import {
	SELECTED_WALLET_COIN,
	SELECTED_WALLET_PAIRED_ACCOUNT,
} from '../../../store/slices/walletSlice';
import SDKService from '../../../services/SDKService';
import { validateDecimals } from '../../../utils/validationsHelper';

const BurnOperation = () => {
	const {
		isOpen: isOpenModalAction,
		onOpen: onOpenModalAction,
		onClose: onCloseModalAction,
	} = useDisclosure();

	const selectedStableCoin = useSelector(SELECTED_WALLET_COIN);
	const account = useSelector(SELECTED_WALLET_PAIRED_ACCOUNT);

	const { decimals = 0, totalSupply } = selectedStableCoin || {};

	const { control, getValues, formState } = useForm({
		mode: 'onChange',
	});

	const { t } = useTranslation(['burn', 'global', 'operations']);

	const handleBurn: ModalsHandlerActionsProps['onConfirm'] = async ({ onSuccess, onError }) => {
		const { amount } = getValues();
		try {
			if (!selectedStableCoin?.memo || !selectedStableCoin?.tokenId) {
				onError();
				return;
			}
			await SDKService.burn({
				proxyContractId: selectedStableCoin.memo,
				account,
				tokenId: selectedStableCoin.tokenId,
				amount,
			});
			onSuccess();
		} catch (error) {
			console.error(error);
			onError();
		}
	};

	return (
		<>
			<OperationLayout
				LeftContent={
					<>
						<Heading data-testid='title' fontSize='24px' fontWeight='700' mb={10} lineHeight='16px'>
							{t('burn:title')}
						</Heading>
						<Text color='brand.gray' data-testid='operation-title'>
							{t('burn:operationTitle')}
						</Text>
						<Stack as='form' spacing={6}>
							<InputNumberController
								rules={{
									required: t('global:validations.required'),
									validate: {
										maxDecimals: (value: number) => {
											return (
												validateDecimals(value, decimals) ||
												t('global:validations.decimalsValidation')
											);
										},
										quantityOverTotalSupply: (value: number) => {
											return (
												(totalSupply && totalSupply >= value) ||
												t('global:validations.overTotalSupply')
											);
										},
									},
								}}
								decimalScale={decimals}
								isRequired
								control={control}
								name='amount'
								label={t('burn:amountLabel')}
								placeholder={t('burn:amountPlaceholder')}
							/>
						</Stack>
					</>
				}
				onConfirm={onOpenModalAction}
				confirmBtnProps={{ isDisabled: !formState.isValid }}
			/>
			<ModalsHandler
				errorNotificationTitle={t('operations:modalErrorTitle')}
				errorNotificationDescription={'error'} // TODO: show returned error from sdk
				modalActionProps={{
					isOpen: isOpenModalAction,
					onClose: onCloseModalAction,
					title: t('burn:modalAction.subtitle'),
					confirmButtonLabel: t('burn:modalAction.accept'),
					onConfirm: handleBurn,
				}}
				ModalActionChildren={
					<DetailsReview
						title={t('burn:modalAction.subtitle')}
						details={[
							{
								label: t('burn:modalAction.amount'),
								value: getValues().amount,
								valueInBold: true,
							},
						]}
					/>
				}
				successNotificationTitle={t('operations:modalSuccessTitle')}
				successNotificationDescription={t('operations:modalSuccessDesc')}
			/>
		</>
	);
};

export default BurnOperation;
