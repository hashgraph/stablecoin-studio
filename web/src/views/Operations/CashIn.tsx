import { Heading, Text, Stack } from '@chakra-ui/react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import DetailsReview from '../../components/DetailsReview';
import InputController from '../../components/Form/InputController';
import InputNumberController from '../../components/Form/InputNumberController';
import { validateAccount } from '../../utils/validationsHelper';
import OperationLayout from './OperationLayout';

const CashInOperation = () => {
	const { control } = useForm({
		mode: 'onChange',
	});
	const { t } = useTranslation(['cashIn', 'global']);

	const handleCashIn = () => {
		// TODO: cashin with sdk
	};

	return (
		<OperationLayout
			LeftContent={
				<>
					<Heading data-testid='title' fontSize='24px' fontWeight='700' mb={10} lineHeight='16px'>
						{t('cashIn:title')}
					</Heading>
					<Text color='brand.gray' data-testid='operation-title'>
						{t('cashIn:operationTitle')}
					</Text>
					<Stack spacing={6}>
						{/* TODO: improve typesctipt  */}
						{/* @ts-ignore */}
						<InputNumberController
							rules={{
								required: t('global:validations.required'),
								// TODO: Add validation of max decimals allowed by stable coin
							}}
							isRequired
							control={control}
							name='amount'
							label={t('cashIn:amountLabel')}
							placeholder={t('cashIn:amountPlaceholder')}
							hasArrows
						/>
						<InputController
							rules={{
								required: t('global:validations.required'),
								validate: {
									validAccount: (value: string) => {
										return validateAccount(value) || t('global:validations.invalidAccount');
									},
								},
							}}
							isRequired
							control={control}
							name='destinationAccount'
							placeholder={t('cashIn:destinationAccountPlaceholder')}
							label={t('cashIn:destinationAccountLabel')}
						/>
					</Stack>
				</>
			}
			RightContent={
				<Stack bg='brand.white' spacing={10}>
					<Heading fontSize='16px' color='brand.secondary' data-testid='details-title'>
						{t('cashIn:details.title')}
					</Heading>

					<DetailsReview
						title={t('cashIn:details.basicTitle')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('cashIn:details.name'),
								value: '',
							},
							{
								label: t('cashIn:details.symbol'),
								value: '',
							},
							{
								label: t('cashIn:details.decimals'),
								value: '',
							},
						]}
					/>
					<DetailsReview
						title={t('cashIn:details.optionalTitle')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('cashIn:details.initialSupply'),
								value: '',
							},
							{
								label: t('cashIn:details.totalSupply'),
								value: '',
							},
							{
								label: t('cashIn:details.supplyType'),
								value: '',
							},
						]}
					/>
				</Stack>
			}
			onConfirm={handleCashIn}
		/>
	);
};

export default CashInOperation;
