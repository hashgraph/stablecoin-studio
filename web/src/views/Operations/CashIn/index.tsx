import { Heading, Text, Stack, useDisclosure } from '@chakra-ui/react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import DetailsReview from '../../../components/DetailsReview';
import InputController from '../../../components/Form/InputController';
import InputNumberController from '../../../components/Form/InputNumberController';
import SDKService from '../../../services/SDKService';
import { validateAccount } from '../../../utils/validationsHelper';
import OperationLayout from './../OperationLayout';
import ModalsHandler from '../../../components/ModalsHandler';
import type { ModalsHandlerActionsProps } from '../../../components/ModalsHandler';
import { useSelector } from 'react-redux';
import { GET_ACK_MESSAGES } from '../../../store/slices/hashpackSlice';

const CashInOperation = () => {
	const {
		isOpen: isOpenModalAction,
		onOpen: onOpenModalAction,
		onClose: onCloseModalAction,
	} = useDisclosure();

	const walletMessages = useSelector(GET_ACK_MESSAGES);

	console.log('Messages in state:', walletMessages);

	const { control, getValues, formState } = useForm({
		mode: 'onChange',
	});

	const { t } = useTranslation(['cashIn', 'global', 'operations']);

	// eslint-disable-next-line @typescript-eslint/no-unused-vars
	const handleCashIn: ModalsHandlerActionsProps['onConfirm'] = async ({ onSuccess, onError }) => {
		const { amount } = getValues();
		try {
			await SDKService.cashIn({
				proxyContractId: '0.0.48261507',
				privateKey:
					'302e020100300506032b6570042204201713ea5a2dc0287b11a6f25a1137c0cad65fb5af52706076de9a9ec5a4b7f625',
				accountId: '0.0.47822430',
				tokenId: '0.0.48261510',
				targetId: '0.0.47822430', // destinationACc
				amount,
			});
			onSuccess();
		} catch (error) {
			console.error(error);
			onError();
		}
	};

	return (
		<>
			<OperationLayout
				LeftContent={
					<>
						<Heading data-testid='title' fontSize='24px' fontWeight='700' mb={10} lineHeight='16px'>
							{t('cashIn:title')}
						</Heading>
						<Text color='brand.gray' data-testid='operation-title'>
							{t('cashIn:operationTitle')}
						</Text>
						<Stack as='form' spacing={6}>
							<InputNumberController
								rules={{
									required: t('global:validations.required'),
									// TODO: Add validation of max decimals allowed by stable coin
								}}
								isRequired
								control={control}
								name='amount'
								label={t('cashIn:amountLabel')}
								placeholder={t('cashIn:amountPlaceholder')}
							/>
							<InputController
								rules={{
									required: t('global:validations.required'),
									validate: {
										validAccount: (value: string) => {
											return validateAccount(value) || t('global:validations.invalidAccount');
										},
									},
								}}
								isRequired
								control={control}
								name='destinationAccount'
								placeholder={t('cashIn:destinationAccountPlaceholder')}
								label={t('cashIn:destinationAccountLabel')}
							/>
						</Stack>
					</>
				}
				onConfirm={onOpenModalAction}
				confirmBtnProps={{ isDisabled: !formState.isValid }}
			/>
			<ModalsHandler
				errorNotificationTitle={t('operations:modalErrorTitle')}
				errorNotificationDescription={'error'} // TODO: show returned error from sdk
				modalActionProps={{
					isOpen: isOpenModalAction,
					onClose: onCloseModalAction,
					title: t('cashIn:modalAction.subtitle'),
					confirmButtonLabel: t('cashIn:modalAction.accept'),
					onConfirm: handleCashIn,
				}}
				ModalActionChildren={
					<DetailsReview
						title={t('cashIn:modalAction.subtitle')}
						details={[
							{
								label: t('cashIn:modalAction.destinationAccount'),
								value: getValues().destinationAccount,
							},
							{
								label: t('cashIn:modalAction.amount'),
								value: getValues().amount,
								valueInBold: true,
							},
						]}
					/>
				}
				successNotificationTitle={t('operations:modalSuccessTitle')}
				successNotificationDescription={t('operations:modalSuccessDesc')}
			/>
		</>
	);
};

export default CashInOperation;
