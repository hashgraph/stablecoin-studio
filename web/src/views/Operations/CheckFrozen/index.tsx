import { useState } from 'react';
import { Heading, Text, Stack, useDisclosure } from '@chakra-ui/react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import DetailsReview from '../../../components/DetailsReview';
import InputController from '../../../components/Form/InputController';
import OperationLayout from '../OperationLayout';
import ModalsHandler from '../../../components/ModalsHandler';
import type { ModalsHandlerActionsProps } from '../../../components/ModalsHandler';
import { handleRequestValidation } from '../../../utils/validationsHelper';
import SDKService from '../../../services/SDKService';
import { SELECTED_WALLET_COIN } from '../../../store/slices/walletSlice';

import { FreezeAccountRequest } from '@hashgraph-dev/stablecoin-npm-sdk';
import { useRefreshCoinInfo } from '../../../hooks/useRefreshCoinInfo';
import { propertyNotFound } from '../../../constant';

const CheckFrozenOperation = () => {
	const {
		isOpen: isOpenModalAction,
		onOpen: onOpenModalAction,
		onClose: onCloseModalAction,
	} = useDisclosure();

	const selectedStableCoin = useSelector(SELECTED_WALLET_COIN);

	const [errorOperation, setErrorOperation] = useState();
	const [errorTransactionUrl, setErrorTransactionUrl] = useState();
	const [request] = useState(
		new FreezeAccountRequest({
			tokenId: selectedStableCoin?.tokenId?.toString() ?? '',
			targetId: '',
		}),
	);

	const { t } = useTranslation(['checkFrozen', 'global', 'operations']);
	const { control, getValues, formState } = useForm({
		mode: 'onChange',
	});

	const [isFrozen, setIsFrozen] = useState(false);

	useRefreshCoinInfo();

	const handleCheckFrozen: ModalsHandlerActionsProps['onConfirm'] = async ({
		onSuccess,
		onError,
		onLoading,
	}) => {
		try {
			onLoading();
			if (!selectedStableCoin?.proxyAddress || !selectedStableCoin?.tokenId?.toString()) {
				onError();
				return;
			}
			const isFrozen: any = await Promise.race([
				SDKService.isAccountFrozen(request),
				new Promise((resolve, reject) => {
					setTimeout(() => {
						reject(new Error("Account frozen status couldn't be obtained in a reasonable time."));
					}, 10000);
				}),
			]).catch((e) => {
				console.log(e.message);
				onOpenModalAction();
				throw e;
			});
			setIsFrozen(isFrozen);
			onSuccess();
		} catch (error: any) {
			setErrorTransactionUrl(error.transactionUrl);
			setErrorOperation(error.message);
			onError();
		}
	};

	return (
		<>
			<OperationLayout
				LeftContent={
					<>
						<Heading data-testid='title' fontSize='24px' fontWeight='700' mb={10} lineHeight='16px'>
							{t('checkFrozen:title')}
						</Heading>
						<Text color='brand.gray' data-testid='operation-title'>
							{t('checkFrozen:operationTitle')}
						</Text>
						<Stack as='form' spacing={6} maxW='520px'>
							<InputController
								rules={{
									required: t('global:validations.required') ?? propertyNotFound,
									validate: {
										validation: (value: string) => {
											request.targetId = value;
											const res = handleRequestValidation(request.validate('targetId'));
											return res;
										},
									},
								}}
								isRequired
								control={control}
								name='targetAccount'
								placeholder={t('checkFrozen:accountPlaceholder') ?? propertyNotFound}
								label={t('checkFrozen:accountLabel') ?? propertyNotFound}
							/>
						</Stack>
					</>
				}
				onConfirm={onOpenModalAction}
				confirmBtnProps={{ isDisabled: !formState.isValid }}
			/>
			<ModalsHandler
				errorNotificationTitle={t('operations:modalErrorTitle')}
				errorNotificationDescription={errorOperation}
				errorTransactionUrl={errorTransactionUrl}
				modalActionProps={{
					isOpen: isOpenModalAction,
					onClose: onCloseModalAction,
					title: t('checkFrozen:modalAction.subtitle'),
					confirmButtonLabel: t('checkFrozen:modalAction.accept'),
					onConfirm: handleCheckFrozen,
				}}
				ModalActionChildren={
					<DetailsReview
						title={t('checkFrozen:modalAction.subtitle')}
						details={[
							{
								label: t('checkFrozen:modalAction.account'),
								value: getValues().targetAccount,
							},
						]}
					/>
				}
				successNotificationTitle={t('operations:modalSuccessTitle')}
				successNotificationDescription={t(
					isFrozen ? 'checkFrozen:modalIsFrozen' : 'checkFrozen:modalIsNotFrozen',
					{
						account: getValues().targetAccount,
					},
				)}
			/>
		</>
	);
};

export default CheckFrozenOperation;
