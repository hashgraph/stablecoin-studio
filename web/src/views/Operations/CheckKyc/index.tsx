import { useState } from 'react';
import { Heading, Text, Stack, useDisclosure } from '@chakra-ui/react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import DetailsReview from '../../../components/DetailsReview';
import InputController from '../../../components/Form/InputController';
import OperationLayout from '../OperationLayout';
import ModalsHandler from '../../../components/ModalsHandler';
import type { ModalsHandlerActionsProps } from '../../../components/ModalsHandler';
import { handleRequestValidation } from '../../../utils/validationsHelper';
import SDKService from '../../../services/SDKService';
import { SELECTED_WALLET_COIN } from '../../../store/slices/walletSlice';

import { KYCRequest } from '@hashgraph/stablecoin-npm-sdk';
import { useRefreshCoinInfo } from '../../../hooks/useRefreshCoinInfo';
import { propertyNotFound } from '../../../constant';

const CheckKycOperation = () => {
	const {
		isOpen: isOpenModalAction,
		onOpen: onOpenModalAction,
		onClose: onCloseModalAction,
	} = useDisclosure();

	const selectedStableCoin = useSelector(SELECTED_WALLET_COIN);

	const [errorOperation, setErrorOperation] = useState();
	const [errorTransactionUrl, setErrorTransactionUrl] = useState();
	const [request] = useState(
		new KYCRequest({
			tokenId: selectedStableCoin?.tokenId?.toString() ?? '',
			targetId: '',
		}),
	);

	const { t } = useTranslation(['checkKyc', 'global', 'operations']);
	const { control, getValues, formState } = useForm({
		mode: 'onChange',
	});

	const [hasKyc, setHasKyc] = useState(false);

	useRefreshCoinInfo();

	const handleCheckKyc: ModalsHandlerActionsProps['onConfirm'] = async ({
		onSuccess,
		onError,
		onLoading,
	}) => {
		try {
			onLoading();
			if (!selectedStableCoin?.proxyAddress || !selectedStableCoin?.tokenId?.toString()) {
				onError();
				return;
			}
			const hasKyc: any = await Promise.race([
				SDKService.isAccountKYCGranted(request),
				new Promise((resolve, reject) => {
					setTimeout(() => {
						reject(new Error("Account KYC information couldn't be obtained in a reasonable time."));
					}, 10000);
				}),
			]).catch((e) => {
				console.log(e.message);
				onOpenModalAction();
				throw e;
			});
			setHasKyc(hasKyc);
			onSuccess();
		} catch (error: any) {
			setErrorTransactionUrl(error.transactionUrl);
			setErrorOperation(error.message);
			onError();
		}
	};

	return (
		<>
			<OperationLayout
				LeftContent={
					<>
						<Heading data-testid='title' fontSize='24px' fontWeight='700' mb={10} lineHeight='16px'>
							{t('checkKyc:title')}
						</Heading>
						<Text color='brand.gray' data-testid='operation-title'>
							{t('checkKyc:operationTitle')}
						</Text>
						<Stack as='form' spacing={6} maxW='520px'>
							<InputController
								rules={{
									required: t('global:validations.required') ?? propertyNotFound,
									validate: {
										validation: (value: string) => {
											request.targetId = value;
											const res = handleRequestValidation(request.validate('targetId'));
											return res;
										},
									},
								}}
								isRequired
								control={control}
								name='targetAccount'
								placeholder={t('checkKyc:accountPlaceholder') ?? propertyNotFound}
								label={t('checkKyc:accountLabel') ?? propertyNotFound}
							/>
						</Stack>
					</>
				}
				onConfirm={onOpenModalAction}
				confirmBtnProps={{ isDisabled: !formState.isValid }}
			/>
			<ModalsHandler
				errorNotificationTitle={t('operations:modalErrorTitle')}
				errorNotificationDescription={errorOperation}
				errorTransactionUrl={errorTransactionUrl}
				modalActionProps={{
					isOpen: isOpenModalAction,
					onClose: onCloseModalAction,
					title: t('checkKyc:modalAction.subtitle'),
					confirmButtonLabel: t('checkKyc:modalAction.accept'),
					onConfirm: handleCheckKyc,
				}}
				ModalActionChildren={
					<DetailsReview
						title={t('checkKyc:modalAction.subtitle')}
						details={[
							{
								label: t('checkKyc:modalAction.account'),
								value: getValues().targetAccount,
							},
						]}
					/>
				}
				successNotificationTitle={t('operations:modalSuccessTitle')}
				successNotificationDescription={t(
					hasKyc ? 'checkKyc:modalHasKyc' : 'checkKyc:modalNotHasKyc',
					{
						account: getValues().targetAccount,
					},
				)}
			/>
		</>
	);
};

export default CheckKycOperation;
