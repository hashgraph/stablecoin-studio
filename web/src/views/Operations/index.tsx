import { Box, Heading } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import BaseContainer from '../../components/BaseContainer';
import { NamedRoutes } from '../../Router/NamedRoutes';
import GridDirectAccess from '../../components/GridDirectAccess';

const Operations = () => {
	const { t } = useTranslation('operations');
	const directAccesses = [
		{
			icon: 'ArrowDown',
			route: NamedRoutes.CashIn,
			title: t('cashInOperation'),
		},
		{
			icon: 'ArrowUp',
			route: NamedRoutes.CashOut,
			title: t('cashOutOperation'),
		},
		{
			icon: 'Coin',
			route: NamedRoutes.Balance,
			title: t('getBalanceOperation'),
		},
		{
			icon: 'ArrowsDownUp',
			route: NamedRoutes.RescueTokens,
			title: t('rescueOperation'),
		},
		{
			icon: 'MinusCircle',
			route: NamedRoutes.Wipe,
			title: t('wipeOperation'),
		},
	];

	return (
		<BaseContainer title={t('title')}>
			<Box p={{ base: 4, md: '128px' }}>
				<Heading fontSize='20px' fontWeight='600' mb={14} data-testid='subtitle'>
					{t('subtitle')}
				</Heading>
				<GridDirectAccess directAccesses={directAccesses} />
			</Box>
		</BaseContainer>
	);
};

export default Operations;
