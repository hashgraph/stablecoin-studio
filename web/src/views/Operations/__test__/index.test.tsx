import { render } from '../../../test/index';
import Operations from '../index';
import translations from '../../../translations/en/operations.json';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

describe(`<${Operations.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<Operations />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should render titles', () => {
		const component = render(<Operations />);

		expect(component.getByTestId('base-container-heading')).toHaveTextContent(translations.title);
		expect(component.getByTestId('subtitle')).toHaveTextContent(translations.subtitle);
	});

	test('should render cashin button', () => {
		const component = render(<Operations />);

		expect(component.getByTestId('direct-access-cashIn')).toHaveTextContent(
			translations.cashInOperation,
		);
	});

	test('should render wipe button', () => {
		const component = render(<Operations />);

		expect(component.getByTestId('direct-access-wipe')).toHaveTextContent(
			translations.wipeOperation,
		);
	});

	test('should render rescue button', () => {
		const component = render(<Operations />);

		expect(component.getByTestId('direct-access-rescue')).toHaveTextContent(
			translations.rescueOperation,
		);
	});

	test('should render burn button', () => {
		const component = render(<Operations />);

		expect(component.getByTestId('direct-access-burn')).toHaveTextContent(
			translations.burnOperation,
		);
	});

	describe('should enable/disable operations depending on selected coin keys', () => {
		const selectedStableCoin = {
			initialSupply: 0,
			tokenId: '0.0.48162226',
			totalSupply: 0,
			name: 'MIDAS',
			symbol: 'MD',
			decimals: 3,
			id: '0.0.48132286',
			maxSupply: '100000',
			treasuryId: '0.0.48160285',
			memo: 'Hedera Accelerator Stable Coin',
			adminKey: {
				key: 'key',
				type: 'ED25519',
			},
			freezeKey: {
				key: 'key',
				type: 'ED25519',
			},
			wipeKey: {
				id: '0.0.48160285',
			},
			supplyKey: {
				id: '0.0.48160285',
			},
		};
		test('should enable cash in', async () => {
			const store = mockStore({
				wallet: {
					data: {
						selectedStableCoin,
					},
				},
			});

			const component = render(<Operations />, store);

			expect(component.getByTestId('direct-access-cashIn')).toBeEnabled();
		});
		test.todo('rest of the tests');
	});
});
