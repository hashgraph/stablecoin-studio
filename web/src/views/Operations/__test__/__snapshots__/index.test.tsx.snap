// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Operations /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-llrpa2"
      data-testid="base-container-heading"
    >
      Operations
    </h2>
    <div
      class="css-1b6lda7"
      data-testid="base-container-children"
    >
      <div
        class="css-src872"
      >
        <h2
          class="chakra-heading css-1brd3j2"
          data-testid="subtitle"
        >
          Create new operation
        </h2>
        <div
          class="css-qbodi0"
        >
          <div
            class="css-180e663"
          >
            <button
              class="css-1g500l7"
              data-testid="direct-access-cashIn"
            >
              <div
                class="css-bv4l9q"
              >
                <svg
                  class="chakra-icon css-1f9qa95"
                  data-testid="direct-access-ArrowDown"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="256"
                    width="256"
                  />
                  <line
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                    x1="128"
                    x2="128"
                    y1="40"
                    y2="216"
                  />
                  <polyline
                    fill="none"
                    points="56 144 128 216 200 144"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                  />
                </svg>
              </div>
              <p
                class="chakra-text css-1mbt7nu"
                data-testid="direct-access-Cash in tokens"
              >
                Cash in tokens
              </p>
            </button>
          </div>
          <div
            class="css-180e663"
          >
            <button
              class="css-kly7w1"
              data-testid="direct-access-wipe"
              disabled=""
            >
              <div
                class="css-bv4l9q"
              >
                <svg
                  class="chakra-icon css-1f9qa95"
                  data-testid="direct-access-MinusCircle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="256"
                    width="256"
                  />
                  <circle
                    cx="128"
                    cy="128"
                    fill="none"
                    r="96"
                    stroke="currentColor"
                    stroke-miterlimit="10"
                    stroke-width="16"
                  />
                  <line
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                    x1="88"
                    x2="168"
                    y1="128"
                    y2="128"
                  />
                </svg>
              </div>
              <p
                class="chakra-text css-1mbt7nu"
                data-testid="direct-access-Wipe tokens"
              >
                Wipe tokens
              </p>
            </button>
          </div>
          <div
            class="css-180e663"
          >
            <button
              class="css-kly7w1"
              data-testid="direct-access-rescue"
              disabled=""
            >
              <div
                class="css-bv4l9q"
              >
                <svg
                  class="chakra-icon css-1f9qa95"
                  data-testid="direct-access-ArrowsDownUp"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="256"
                    width="256"
                  />
                  <polyline
                    fill="none"
                    points="112 176 80 208 48 176"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                  />
                  <line
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                    x1="80"
                    x2="80"
                    y1="48"
                    y2="208"
                  />
                  <polyline
                    fill="none"
                    points="144 80 176 48 208 80"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                  />
                  <line
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                    x1="176"
                    x2="176"
                    y1="208"
                    y2="48"
                  />
                </svg>
              </div>
              <p
                class="chakra-text css-1mbt7nu"
                data-testid="direct-access-Rescue tokens"
              >
                Rescue tokens
              </p>
            </button>
          </div>
          <div
            class="css-180e663"
          >
            <button
              class="css-kly7w1"
              data-testid="direct-access-cashOut"
              disabled=""
            >
              <div
                class="css-bv4l9q"
              >
                <svg
                  class="chakra-icon css-1f9qa95"
                  data-testid="direct-access-ArrowUp"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="256"
                    width="256"
                  />
                  <line
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                    x1="128"
                    x2="128"
                    y1="216"
                    y2="40"
                  />
                  <polyline
                    fill="none"
                    points="56 112 128 40 200 112"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                  />
                </svg>
              </div>
              <p
                class="chakra-text css-1mbt7nu"
                data-testid="direct-access-Cash out"
              >
                Cash out
              </p>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
